src/message_router/router.rs

```diff
@@ -261,9 +226,8 @@ fn create_not_understood_response(
 ) -> Result<FipaMessage, RouterError> {
     // Convert error content to MessageContent with proper validation
     let message_content = MessageContent::try_new(error_content.into_bytes()).map_err(|_| {
-        RouterError::ValidationError {
-            field: FIELD_ERROR_CONTENT.to_string(),
-            reason: REASON_ERROR_CONTENT_TOO_LARGE.to_string(),
+        RouterError::ContentSizeError {
+            message: "Generated error content exceeds maximum message size limit".to_string(),
         }
     })?;
```

Comment:
Why can't we just use the error returned from `MessageContent::try_new`? We might need to adapt it using the tools in thiserror.